src/crypto/crypto.ts

```diff
@@ -52,12 +53,26 @@ const keychainPromises = {
    * @param account The keychain account name.
    */
   getPassword(_keychain: KeyChain, service: string, account: string): Promise<CredType> {
-    return new Promise((resolve, reject): {} =>
-      _keychain.getPassword({ service, account }, (err: Nullable<Error>, password?: string) => {
-        if (err) return reject(err);
-        return resolve({ username: account, password: ensure(password) });
-      })
-    );
+    const sb = Cache.get<SecureBuffer<string>>(`${service}:${account}`);
+    if (!sb) {
+      return new Promise((resolve, reject): {} => {
+        return _keychain.getPassword({ service, account }, (err: Nullable<Error>, password?: string) => {
+          if (err) return reject(err);
+          const newSb = new SecureBuffer<string>();
+          newSb.consume(Buffer.from(ensure(password), 'utf8'));
```

Comment:
At this moment, password has been properly fetch from the SecureBuffer used as part of the keychain access. The password will need to be stored in the cache in a SecureBuffer, so create SecureBuffer and have buffer consume the password, then store it in cache.